src/components/ShareStructure/ShareStructure.vue

```diff
@@ -1,427 +1,731 @@
+    type: 'Class',
+    name: '',
+    hasMaximumShares: true,
+    maxNumberOfShares: null,
+    hasParValue: true,
+    parValue: null,
+    currency: 'CAD',
+    hasRightsOrRestrictions: false,
+    series: [],
+    action: ActionTypes.ADDED
   }
 
-  private getMaximumShareRule (): Array<Function> {
-    let rules: Array<Function> = []
-    if (!this.hasNoMaximumShares) {
-      rules = [
-        v => !!v || 'Maximum share value is required',
-        v => /^\d+$/.test(v) || 'Must be a number greater than 0']
-      // To prevent changing share class value to a lower value after adding series.
-      if (this.isClass && this.activeIndex !== -1 && !this.hasNoMaximumShares &&
-       this.shareStructure.series.length > 0) {
-        const seriesSum = this.shareStructure.series.reduce((a, b) => +a + +b.maxNumberOfShares, 0)
-        rules.push(v => +v >= seriesSum ||
-        'The number for the series (or all series combined, if there are multiple under ' +
-        'a class) cannot exceed the number for the class')
-      }
-      if (this.isSeries && this.shareClasses[this.parentIndex].hasMaximumShares) {
-        let filteredSeries = this.shareClasses[this.parentIndex].series
-        // The series is in edit mode and should be avoided from the calculating the total
-        if (this.activeIndex !== -1) {
-          filteredSeries = filteredSeries.filter((series) => (
-            series.id !== this.shareClasses[this.parentIndex].series[this.activeIndex].id))
-        }
-        const currentSum = filteredSeries.reduce((a, b) => +a + +b.maxNumberOfShares, 0)
-        rules.push(v => +v + currentSum <= +this.shareClasses[this.parentIndex].maxNumberOfShares ||
-        'The number for the series (or all series combined, if there are multiple under ' +
-        'a class) cannot exceed the number for the class')
-      }
-    }
-    return rules
+  private newShareSeries: ShareClassIF = {
+    id: null,
+    priority: null,
+    type: 'Series',
+    name: '',
+    hasMaximumShares: true,
+    maxNumberOfShares: null,
+    hasParValue: true,
+    parValue: null,
+    currency: null,
+    hasRightsOrRestrictions: false,
+    action: ActionTypes.ADDED
   }
 
-  private getParValueRule (): Array<Function> {
-    if (!this.hasNoParValue) {
-      return [
-        v => !!v || 'Par value is required',
-        v => v > 0 || 'Amount must be greater than 0',
-        v => (v < 1) ? (/^(\d+(\.\d{0,3})?|\.\d{0,3})$/.test(v) || 'Amounts less than 1 can be entered with up to 3 decimal place')
-          : (/^\d+(\.\d{1,2})?$/.test(v) || 'Amounts greater than 1 can be entered with up to 2 decimal place')]
-    }
-    return []
+  // Share Class Functionality
+  /**
+   * Initialize the Add Share Class Form
+   */
+  private initNewShareClass (): void {
+    this.currentShareStructure = { ...this.newShareClass }
+    this.currentShareStructure.priority =
+      this.shareClasses.length === 0 ? 1 : this.shareClasses[this.shareClasses.length - 1].priority + 1
```

Comment:
The conditional (ternary) operator is supposed to make simple code easier to read.
Maybe here an if-then-else would be easier to read.
Or parentheses around the if and else bits?

Ditto on line 358 below.

Another option is to separate this into 3 lines, eg:
```
val = condition
   ? if_truthy
   : if_falsy
```

Reply:
Agreed, it was lengthy! Updated in commit